### PR TITLE
[bitnami/zookeeper] Lint chart and standardize

### PR DESCRIPTION
**Pull request by javsalgar**
Signed-off-by: Javier J. Salmeron Garcia <jsalmeron@vmware.com>

**Description of the change**

**Benefits**

**Possible drawbacks**

**Applicable issues**

  - fixes #

**Additional information**

**Checklist** 
- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)
- [x] If the chart contains a `values-production.yaml` apart from `values.yaml`, ensure that you implement the changes in both files

:warning: Keep in mind that if you want to make changes to the kubeapps chart, please implement them in the [kubeapps repository](https://github.com/kubeapps/kubeapps/tree/master/chart/kubeapps). This is only a synchronized mirror.

### CODE REVIEW

File: bitnami/zookeeper/templates/metrics-deployment.yaml [hunk 1]
```diff
@@ -3,66 +3,62 @@ apiVersion: apps/v1
...
+            {{- $servicePort := int .Values.service.port }}
+            {{- $releaseNamespace := .Release.Namespace }}
+            {{- $zookeeperFullname := include "zookeeper.fullname" . }}
+            {{- $zookeeperHeadlessServiceName := printf "%s-%s" $zookeeperFullname "headless" | trunc 63  }}
+            {{- $clusterDomain := .Values.clusterDomain }}
+            - {{range $i, $e := until $replicaCount }}{{ $zookeeperFullname }}-{{ $e }}.{{ $zookeeperHeadlessServiceName }}.{{ $releaseNamespace }}.svc.{{ $clusterDomain }}:{{ $servicePort }}{{- if (lt $e ( sub $replicaCount 1)) -}},{{- end -}}{{ end }}
+          ports:
+            - name: metrics
+              containerPort: 8080
+          resources: {{- toYaml .Values.metrics.resources | nindent 12 }}
```
Comment (juan131) [inline bitnami/zookeeper/templates/metrics-deployment.yaml]: Suggestion:
```diff
+         {{- if .Values.metrics.resources }}
          resources: {{- toYaml .Values.metrics.resources | nindent 12 }}
+         {{- end }}
```

File: bitnami/zookeeper/templates/statefulset.yaml [hunk 2]
```diff
@@ -21,161 +17,160 @@ spec:
...
-        volumeMounts:
-        - name: data
-          mountPath: /bitnami/zookeeper
+        - name: volume-permissions
+          image: {{ template "zookeeper.volumePermissions.image" . }}
+          imagePullPolicy: {{ default "" .Values.volumePermissions.image.pullPolicy | quote }}
+          command: ["chown", "-R", "{{ .Values.securityContext.runAsUser }}:{{ .Values.securityContext.fsGroup }}", "/bitnami/zookeeper"]
+          securityContext:
+            runAsUser: 0
+          resources: {{- toYaml .Values.volumePermissions.resources | nindent 12 }}
```
Comment (juan131) [inline bitnami/zookeeper/templates/statefulset.yaml]: Suggestion:
```diff
+         {{- if .Values.volumePermissions.resources}}
          resources: {{- toYaml .Values.volumePermissions.resources | nindent 12 }}
+         {{- end }}
```

File: bitnami/zookeeper/templates/statefulset.yaml [hunk 3]
```diff
@@ -21,161 +17,160 @@ spec:
...
+                HOSTNAME=`hostname -s`
+                if [[ $HOSTNAME =~ (.*)-([0-9]+)$ ]]; then
+                  ORD=${BASH_REMATCH[2]}
+                  export ZOO_SERVER_ID=$((ORD+1))
+                else
+                  echo "Failed to get index from hostname $HOST"
+                  exit 1
+                fi
+                exec /entrypoint.sh /run.sh
+          resources: {{- toYaml .Values.resources | nindent 12 }}
```
Comment (juan131) [inline bitnami/zookeeper/templates/statefulset.yaml]: Suggestion:
```diff
+         {{- if .Values.resources}}
          resources: {{- toYaml .Values.resources | nindent 12 }}
+         {{- end }}
```

File: bitnami/zookeeper/values.yaml [hunk 4]
```diff
@@ -248,17 +248,28 @@ metrics:
...
+
+  ## Tolerations for pod assignment
+  ## Ref: https://kubernetes.io/docs/concepts/configuration/taint-and-toleration/
+  ##
+  tolerations: []
+
+  ## Affinity for pod assignment
+  ## Ref: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#affinity-and-anti-affinity
+  ##
+  affinity: {}
```
Comment (juan131) [inline bitnami/zookeeper/values.yaml]: Did you miss this block in **values-production.yaml**?

Comment (javsalgar) [reply bitnami/zookeeper/values.yaml]: FIxed

**Pull request merged**